### PR TITLE
MWPW-135374 Blog Bulk Update Boilerplate

### PR DESCRIPTION
**Pull request by Brandon32**
* Added Boilerplate for blog to milo migration
* Refactored code and created utils/fetch-utils.js
* Added TODOs in blog-migration.js for upcoming tickets

Resolves: [MWPW-135374](https://jira.corp.adobe.com/browse/MWPW-135374)

### CODE REVIEW

File: block-report.js [hunk 1]
```diff
@@ -99,7 +56,7 @@ async function runReport(project, site, indexUrl, cached = true) {
             totals.blocks[blockName] = (totals.blocks[blockName] || 0) + 1;
 
             const links = getNodesByType(table.table, 'link');
-            const name = entry + ',' + blockName + j;
+            const name = entry + ',' + blockName + ',' + j;
```
Comment (meganthecoder) [inline block-report.js:59]: Nit: template string
```suggestion
            const name = `${entry},${blockName},${j}`;
```

File: utils/fetch-utils.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import { writeFile, readFile, mkdir } from 'fs/promises';
+import { fetch } from '@adobe/fetch';
+
+const delay = (milliseconds) => new Promise((resolve) => setTimeout(resolve, milliseconds));
+
+async function fetchIndex(url) {
+    console.log('Fetching entries and saving locally');
+    const index = await fetch(url);
```
Comment (meganthecoder) [inline utils/fetch-utils.js:8]: Does this need error handling?
```suggestion
    const index = await fetch(url);
    if (!index.ok) throw new Error(`Error fetching index: ${index.status}`);
```

**Review by meganthecoder — APPROVED** [review]
Looks good! Just a couple nits

File: utils/fetch-utils.js (hunk 2 above)
Comment (Brandon32) [reply utils/fetch-utils.js:8]: That's a good idea, thanks

**Review by JasonHowellSlavin — COMMENTED** [review]
Looks Great. I see a lot of `project, site, index, cached` as params being passed. Might be worth it to think of using a config to avoid passing same / similar args?

File: blog-migration.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            const mdast = await getMdast(markdown);
+
+            // TODO: Migration Part 1 - Pull Quote
+
+            // TODO: Migration Part 2 - Embed
+
+            // TODO: Migration Part 3 - Images
+
+            const outputFolder = `${entry.replace(mdDir, outputDir).split('/').slice(0, -1).join('/')}`;
+            const outputFile = `${entry.split('/').slice(-1)[0].split('.').slice(0, -1).join('.')}.docx`;
```
Comment (JasonHowellSlavin) [inline blog-migration.js:81]: maybe could add a comment on what the expected string is from these methods?

**Review by JasonHowellSlavin — APPROVED** [review]

**Comment by Brandon32** [comment]
> I see a lot of project, site, index, cached as params being passed. Might be worth it to think of using a config to avoid passing same / similar args?
@JasonHowellSlavin I'm not sure a config makes sense to me, I don't want to create an object just to de-structure it but if it's just being passed through then it makes sense. If there is a specific place then let me know, but with this PR there is parameter repetition because there are 3 different scripts.

File: blog-migration.js (hunk 3 above)
Comment (Brandon32) [reply blog-migration.js:81]: Good call, this is a little convoluted, plus we don't need the templating on `outputFolder`. 
I'll break this down for clarity.

**Pull request merged**